codegen/src/attribute.rs

```diff
@@ -129,30 +130,50 @@ impl Step {
                     ::std::boxed::Box::pin(f)
                 }
 
-                #caller_name
+                let f: ::cucumber::Step<#world> = #caller_name;
+                f
             }
         };
 
         Ok(quote! {
             #func
 
             #[automatically_derived]
-            ::cucumber::codegen::submit!(
-                #![crate = ::cucumber::codegen] {
-                    <#world as ::cucumber::codegen::WorldInventory<
-                        _, _, _,
-                    >>::#constructor_method(
-                        ::cucumber::step::Location {
-                            path: ::std::convert::From::from(::std::file!()),
-                            line: ::std::line!(),
-                            column: ::std::column!(),
-                        },
-                        ::cucumber::codegen::Regex::new(#step_matcher)
-                            .unwrap(),
-                        #step_caller,
-                    )
+            ::cucumber::codegen::submit!({
+                // TODO: remove this, once `#![feature(more_qualified_paths)]`
+                //       is stabilized.
+                //       https://github.com/rust-lang/rust/issues/86935
+                type StepAlias =
+                    <#world as ::cucumber::codegen::WorldInventory>::#step_type;
+
+                StepAlias {
+                    loc: ::cucumber::step::Location {
+                        path: ::std::file!(),
+                        line: ::std::line!(),
+                        column: ::std::column!(),
+                    },
+                    regex: {
+                        let lazy: ::cucumber::codegen::LazyRegex = || {
+                            static LAZY: ::cucumber::codegen::Lazy<
+                                ::cucumber::codegen::Regex
+                            > = ::cucumber::codegen::Lazy::new(|| {
+                                ::cucumber::codegen::Regex::new(#step_matcher)
+                                    .unwrap()
+                            });
+                            LAZY.clone()
+                        };
```

Comment:
@ilslv what is the motivation with it being a function? Why we cannot just return `&'static Regex` and do the `.clone()` inside `WorldInit::collection()`?

Reply:
Ahhh, yeah, `static LAZY: ::cucumber::codegen::Lazy<_>` is not `const` constructible.